web/src/app/modules/navigation/modules/navigator/services/navigator.service.ts

```diff
@@ -18,20 +19,26 @@ export class NavigatorService {
         private dataService: SpecmateDataService,
         private logger: LoggingService,
         private router: Router,
-        private route: ActivatedRoute) {
+        private route: ActivatedRoute,
+        private location: Location) {
+
+        this.location.subscribe(pse => {
```

Comment:
Would it be possible to handle all of this stuff with the location, and just omit the ActivatedRoute? I never really understood this anyways.

Reply:
I'm not sure we can. 
We use route to get the currentUrl, which looks for example as follows:
`test-data/Folder-1/Requirement-4`
From location, we can get the path, which looks for example as follows:
`/-/Requirement/test-data%2FFolder-1%2FRequirement-4`

I can urldecode it, but then we still have the basePath which is constructed in `url.ts`. I can implement there a function stripBasePath(path) which would return the url, needed for the dataservice to return the correct element. Would that work?

Reply:
Yes, that should work.